File: lista_de_exercicios/questao15/questao15.py
class SaldoInsuficienteException(Exception):
    pass

class ContaBancaria:
    def __init__(self, saldo):
        self.saldo = saldo

    def sacar(self, valor):
        if valor > self.saldo:
            raise SaldoInsuficienteException("Saldo insuficiente para o saque.")
        self.saldo -= valor

File: lista_de_exercicios/questao15/questao15.go
package main

import "errors"

type SaldoInsuficienteException struct {
    message string
}

func (e *SaldoInsuficienteException) Error() string {
    return e.message
}

type ContaBancaria struct {
    saldo float64
}

func NewContaBancaria(saldo float64) *ContaBancaria {
    return &ContaBancaria{saldo: saldo}
}

func (c *ContaBancaria) Sacar(valor float64) error {
    if valor > c.saldo {
        return &SaldoInsuficienteException{message: "Saldo insuficiente para o saque."}
    }
    c.saldo -= valor
    return nil
}
